Mecha Tourian: fix FX1 paletteflags

jedat555 opened the pull request:
Similar to one of my previous Red Brinstar PRs, fixes FX1 paletteflags in various rooms.

jedat555 committed:
mecha tourian fx1 fixes

amoebaOfDoom commented:
Seems like there's a bunch of other changes mixed in here that aren't described.

jedat555 commented:
> Seems like there's a bunch of other changes mixed in here that aren't described.

The additional changes in BT's room are very minor and purely stylistic: I swapped out two tiles around the door because I didn't really like how they looked (since this is a room I retiled, I figured doing that wouldn't be an issue).

Switching Spazer Beam Room to use Layer2 instead of BGData was on the other hand a change I made because I only realised after already adding FX1s to that room that the BGData version of the background doesn't actually animate, so I decided to just switch it to Layer2 and swap out those red and blue panel tiles with their alternative versions which do animate and which also are used in certain other rooms in this theme. If this is an undesirable change it can easily be reverted.

That should be it as far as non-disclosed changes go, though I suppose it is worth pointing out that I did also add FX1s to the four vanilla escape rooms - I figured that would make sense to do, since unlike in vanilla where the other escape effects render the basic glow obsolete, in map rando these rooms are mostly encountered outside of that context so it's weird for the glow effect to just be absent. This can also easily be reverted though, of course.

kjbranch commented:
palette flags all look to be improvements except I wasnt sure on the following:
- don't see a difference in: hatchling container, old tourian shaft, silver torizo (animals), west cac alley, spring ball
- colosseum: there is a lamp on the top left floating platform that is not glowing with these changes (not sure what was supposed to be changed?)

note that i didnt test all of them, especially the 0x00 -> 0x02 (or similar)

non-palette changing notes:
- spazer: not sure i fully understand the change, but it does fix the background glow, as you mentioned
- ST: looks good

jedat555 committed:
Merge branch 'amoebaOfDoom:main' into small_fixes

jedat555 committed:
undo colosseum change

jedat555 commented:
> palette flags all look to be improvements except I wasnt sure on the following:
> 
>     * don't see a difference in: hatchling container, old tourian shaft, silver torizo (animals), west cac alley, spring ball
> 
>     * colosseum: there is a lamp on the top left floating platform that is not glowing with these changes (not sure what was supposed to be changed?)
> 
> 
> note that i didnt test all of them, especially the 0x00 -> 0x02 (or similar)
> 
> non-palette changing notes:
> 
>     * spazer: not sure i fully understand the change, but it does fix the background glow, as you mentioned
> 
>     * ST: looks good

- hatchling container just removes some unneeded fx1 flags leftover from its vanilla theme (stuff to handle the heat glow); old tourian shaft turns the lights off for the asleep state of the room to match other such rooms in the theme; ST i'm not sure, i think i changed it because the escape fx were already achieving the same effect (i can revert this change if needed); west cac alley doesn't have any glowy bits, so i just got rid of the fx1 flag that wasn't doing anything; spring ball is a similar case, except with the sand
- i must've overlooked the one lamp in colosseum, so i reverted that change

kjbranch commented:
Ah, everything makes sense now (I was staring at the hatchling room for at least 5 mins haha).
Old tourian i wouldve realized if i went back to it again at the end. 

This all looks good to me now and I think it can be merged as well (someone more knowledgeable might want to go through it. Though, I think I understand what is going on now, and I did test a large number of them and they all look good).

jedat555 committed:
undo spazer room change

jedat555 commented:
Reverted the spazer beam room change to hopefully clean up the PR.
